Notebooks/gesla.py

```diff
@@ -0,0 +1,190 @@
+        metadata for each record.
+
+        Args:
+            lat (float): latitude on the interval [-90, 90]
+            lon (float): longitude on the interval [-180, 180]
+            N (int, optional): number of locations to load. Defaults to 1.
+            force_xarray (bool, optional): if N=1, the default behavior is to
+                return a pandas.DataFrame object containing data/flags and a
+                pandas.Series object containing metadata. Set this argument to
+                True to return a xarray Dataset even if N=1. Defaults to False.
+
+        Returns:
+            xarray.Dataset: data, flags, and metadata for each record.
+        """
+        N = int(N)
+        if N <= 0:
+            raise Exception("Must specify N > 0")
+
+        d = (self.meta.longitude - lon) ** 2 + (self.meta.latitude - lat) ** 2
+        idx = d.sort_values().iloc[:N].index
+        meta = self.meta.loc[idx]
+
+        if (N > 1) or force_xarray:
+            return self.files_to_xarray(meta.filename.tolist())
+
+        else:
+            data, meta = self.file_to_pandas(meta.filename.values[0])
+            return data, meta
+
+    def load_lat_lon_range(
+        self,
+        region: Optional[Union[Polygon, MultiPolygon]] = None,
+        south_lat : Optional[float] = None,
+        north_lat : Optional[float] = None,
+        west_lon : Optional[float] = None,
+        east_lon : Optional[float] = None,
+        force_xarray=False,
+    ):
+        """Load GESLA records within a rectangular lat/lon range into a xarray.
+        Dataset object.
+
+        Args:
+            south_lat (float, optional): southern extent of the range. Defaults
+                to -90.
+            north_lat (float, optional): northern extent of the range. Defaults
+                to 90.
+            west_lon (float, optional): western extent of the range. Defaults
+                to -180.
+            east_lon (float, optional): eastern extent of the range. Defaults
+                to 180.
+            force_xarray (bool, optional): if there is only one record in the
+                lat/lon range, the default behavior is to return a
+                pandas.DataFrame object containing data/flags and a
+                pandas.Series object containing metadata. Set this argument to
+                True to return a xarray.Dataset even if only one record is
+                selected. Defaults to False.
+
+        Returns:
+            xarray.Dataset: data, flags, and metadata for each record.
+        """
+        region = get_region(
+            region=region,
+            lon_min=west_lon,
+            lon_max=east_lon,
+            lat_min=south_lat,
+            lat_max=north_lat,
+            symmetric=True,
+        )
+
+        locations = gp.GeoDataFrame(
+            self.meta, geometry=gp.points_from_xy(self.meta.longitude, self.meta.latitude), crs="EPSG:4326"
+        )
```

Comment:
would it make sense to have this directly in `__init__`?

Reply:
It does make sense. 
I merged gesla.py function from @philiprt and the regional extraction from searvey.

I proposed a pull request if you want more details of what is theirs and what is searvey's : https://github.com/philiprt/GeslaDataset/pull/3/files

Reply:
Cool.

For the record, the `lon_min`, `lon_max` etc are probably not that good an idea after all. They are needlessly complicating the API It is trivial to construct a BBox using `shapely.box()` function, but I didn't know about this function at the time.